spmd/tensor/ops/math_ops.py

```diff
@@ -117,24 +117,37 @@ def einop_rule(
     )
 
 
-@register_impl("aten.sum.default")
-def dist_sum(self: DTensor) -> DTensor:
-    self_local = self.to_local()
-    self_placement = self.placements[0]
-    device_mesh = self.device_mesh
-
-    local_sum = self_local.sum()
-
-    if self_placement.is_shard() or self_placement.is_partial():
-        placements = [_Partial()]
-        # partial reduce
-        partial_sum = DTensor(local_sum, device_mesh, placements)
-        # all_reduce across device
-        replicate_placements = [Replicate()]
-        return partial_sum.redistribute(device_mesh, replicate_placements)
-    elif self_placement.is_replicate():
-        return DTensor(
-            local_sum, device_mesh=device_mesh, placements=self.placements
+def reduction_rule(op_schema: OpSchema) -> OutputSharding:
+    """
+    Propagate the sharding for reduction operations. Examples:
+        ij->i - sum on dim
+    """
+    alphabet = "abcdefghijklmnopqrstuvwxyz"
+    # reduction op usually begin with a single tensor
+    input_spec = cast(PlacementSpec, op_schema.args_schema[0])
```

Comment:
This is actually the spec of the input Distributed Tensor, right?

Reply:
Yep